intake_bluesky/core.py

```diff
@@ -889,3 +891,101 @@ def parse_handler_registry(handler_registry):
 
 intake.registry['remote-bluesky-run'] = RemoteBlueskyRun
 intake.container.container_map['bluesky-run'] = RemoteBlueskyRun
+
+
+def concat_dataarray_pages(dataarray_pages):
+    """
+    Combines a iterable of dataarray_pages to a single dataarray_page.
+
+    Parameters
+    ----------
+    dataarray_pages: Iterabile
+        An iterable of event_pages with xarray.dataArrays in the data,
+        timestamp, and filled fields.
+    Returns
+    ------
+    event_page : dict
+        A single event_pages with xarray.dataArrays in the data,
+        timestamp, and filled fields.
+    """
+    pages = list(dataarray_pages)
+    if len(pages) == 1:
+        return pages[0]
+
+    array_keys = ['seq_num', 'time', 'uid']
+    data_keys = dataarray_pages[0]['data'].keys()
+
+    return {'descriptor': pages[0]['descriptor'],
+            **{key: list(itertools.chain.from_iterable(
+                    [page[key] for page in pages])) for key in array_keys},
+            'data': {key: xarray.concat([page['data'][key] for page in pages])
+                     for key in data_keys},
+            'timestamps': {key: xarray.concat([page['timestamps'][key] for page in pages])
+                           for key in data_keys},
+            'filled': {key: xarray.concat([page['filled'][key] for page in pages])
+                       for key in data_keys}}
+
+
+def event_page_to_dataarray_page(event_page, dims=None, coords=None):
+    """
+    Converts the event_page's data, timestamps, and filled to xarray.DataArray.
+
+    Parameters
+    ----------
+    event_page: dict
+    Returns
+    ------
+    event_page : dict
+        An event_pages with xarray.dataArrays in the data,
+        timestamp, and filled fields.
+    """
+    if coords is None:
+        coords = {'time': event_page['time']}
+    if dims is None:
+        dims = ('time',)
+
+    for key in event_page['data']:
```

Comment:
Can this work on a copy instead of modifying in place? (None of the large arrays should be actually copied.)

Reply:
updated